### PR TITLE
Python bindings

### PR DESCRIPTION
**Pull request by ftalbrecht**
This introduces python bindings, see https://github.com/dune-community/dune-xt-common/pull/24 and https://github.com/dune-community/dune-xt-grid/pull/14. Things we might want to change (though not necessarily in this pr):
- Add a good way to (not manually) add functions for all grids/dimension, etc...

**Before we merge** we also need to revert e5d79201cfabeef546739c05fa3eb560bbda4c14.

### CODE REVIEW

File: dune/xt/CMakeLists.txt [hunk 1]
```diff
@@ -9,4 +9,9 @@ add_subdirectory(functions)
 
 add_subdirectory(functions/test EXCLUDE_FROM_ALL)
 
+if(dune-pybindxi_FOUND)
+  dune_pybindxi_add_module(functions functions/bindings.cc)
+  target_link_dune_default_libraries(functions)
```
Comment (renefritze) [inline dune/xt/CMakeLists.txt]: I think this shouldn't be necessary when using the DUNE's simple cmake mode properly in the `dune_pybindxi_add_module` macro

Comment (ftalbrecht) [reply dune/xt/CMakeLists.txt]: You mean the `target_link_dune_default_libraries` part?

Comment (renefritze) [reply dune/xt/CMakeLists.txt]: Yeah. That should automatically if the target was added as a `dune_add_library`

**Comment by ftalbrecht** [comment]
Done in 25d226e87fbf70d6ae3bfd4cf7ab9f5d35aa3f16.

**Pull request closed without merging**